WcaOnRails/app/models/championship.rb

```diff
@@ -15,16 +16,46 @@ class Championship < ApplicationRecord
                                 inclusion: { in: CHAMPIONSHIP_TYPES }
 
   def name
-    return "World Championship" if championship_type == "world"
+    return I18n.t('competitions.competition_form.championship_types.world') if championship_type == "world"
 
-    return "Greater China Championship" if championship_type == "greater_china"
+    return I18n.t('competitions.competition_form.championship_types.greater_china') if championship_type == "greater_china"
 
     continent = Continent.c_find(championship_type)
-    return "Continental Championship for #{continent.name}" if continent
+    return I18n.t('competitions.competition_form.championship_types.continental', continent: continent.name) if continent
 
     country = Country.find_by_iso2(championship_type)
-    return "National Championship for #{country.name}" if country
+    return I18n.t('competitions.competition_form.championship_types.national', country: country.name) if country
 
-    "Championship for #{championship_type}"
+    I18n.t('competitions.competition_form.championship_types.generic', type: championship_type)
+  end
+
+  def national?
+    Country.find_by_iso2(championship_type)
+  end
+
+  def greater_china?
+    championship_type == "greater_china"
+  end
+
+  def continental?
+    Continent.c_find(championship_type)
+  end
+
+  def world?
+    championship_type == "world"
+  end
+
+  def <=>(other)
```

Comment:
I always find these sorts of methods impossible to read when written this way. A pattern I've been very fond of is to munge the object into an array, which you can then compare lexicographically. See https://stackoverflow.com/a/28558053/1739415 for an example of this. In this case, I think we'd do something like this:

```ruby
def to_a
  [
    world? ? 0 : 1,
    continental? ? 0 : 1,
    greater_china? ? 0 : 1,
    national? ? 0 : 1,
  ]
end

def <=>(other)
  self.to_a <=> other.to_a
end
```

This pattern may feel a bit weird at first, but I like it because it's both explicit and concise about what attributes take precedence over what other things.

Reply:
Wow, this is super legible. I don't think I've ever seen a particularly legible `compare_to` function that wasn't trivial. I'll definitely do this, thanks!